### DISCUSSION TITLE
Using Foxglove Studio's mcap with fast-dds

### DISCUSSION THREAD

**Discussion by samehmohamed88** [eProsima/Fast-DDS#2966] [Q&A]
[Foxglove](https://foxglove.dev/) is a very nice alternative to rviz2.  Chief among its benefits is there's no dependency on the ros ecosystem, so as long as you have Topics and data is some [predefined format ](https://github.com/foxglove/mcap), it can visualize it in some pretty good ways.

Unfortunately their mcap format does not support Fast-CDR's IDL, but they do support protobuf.  I am trying to bridge the best of both worlds, where I can use Fast-DDS with eprosima [fast buffers](https://www.eprosima.com/index.php/resources-all/performance/apache-thrift-vs-protocol-buffers-vs-fast-buffers), but somehow either in an offline or background process, log message data to protobuf for consumption in Foxglove.

1. Is there a simple Fast-CDR IDL to protobuf conversion tool out there?
2. Beyond rviz2 are you aware of data visualization tools that support Fast-CDR IDL?

**Comment by EduPonz** [comment]
Hi @armuren ,

I guess you could try to use ros2 bag (which supports mcap) as we describe [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/rosbag_capture/rosbag_capture.html#rosbag-capture). Else, you may be interested in [Integration Service](https://github.com/eProsima/Integration-Service), a tool to bridge communication between different protocols. However, we currently do not have a system handle for protobuf, so you'd need to do some coding.

Regarding data visualization, both [webots](https://cyberbotics.com/) and and [PlotJuggler](https://www.plotjuggler.io/) (through the [Fast DDS Visualizer Plugin](https://github.com/eProsima/fastdds-visualizer-plugin)) support Fast DDS.
